addons/mail/static/src/widgets/messaging_menu/messaging_menu.js

```diff
@@ -1,50 +1,26 @@
 /** @odoo-module **/
 
-import { MessagingMenu } from '@mail/components/messaging_menu/messaging_menu';
-
-import SystrayMenu from 'web.SystrayMenu';
-import Widget from 'web.Widget';
+import { registry } from "@web/core/registry";
 
-/**
- * Odoo Widget, necessary to instantiate component.
- */
-const MessagingMenuWidget = Widget.extend({
-    template: 'mail.widgets.MessagingMenu',
-    /**
-     * @override
-     */
-    init() {
-        this._super(...arguments);
-        this.component = undefined;
-    },
-    /**
-     * @override
-     */
-    destroy() {
-        if (this.component) {
-            this.component.destroy();
-        }
-        this._super(...arguments);
-    },
-    async on_attach_callback() {
-        this.component = new MessagingMenu(null);
-        await this.component.mount(this.el);
-        // unwrap
-        this.el.parentNode.insertBefore(this.component.el, this.el);
-        this.el.parentNode.removeChild(this.el);
-    },
-});
+import { MessagingMenu } from '@mail/components/messaging_menu/messaging_menu';
 
 // Systray menu items display order matches order in the list
 // lower index comes first, and display is from right to left.
 // For messagin menu, it should come before activity menu, if any
 // otherwise, it is the next systray item.
-const activityMenuIndex = SystrayMenu.Items.findIndex(SystrayMenuItem =>
-    SystrayMenuItem.prototype.name === 'activity_menu');
-if (activityMenuIndex > 0) {
-    SystrayMenu.Items.splice(activityMenuIndex, 0, MessagingMenuWidget);
-} else {
-    SystrayMenu.Items.push(MessagingMenuWidget);
-}
+// const activityMenuIndex = SystrayMenu.Items.findIndex(SystrayMenuItem =>
+//     SystrayMenuItem.prototype.name === 'activity_menu');
+// if (activityMenuIndex > 0) {
+//     SystrayMenu.Items.splice(activityMenuIndex, 0, MessagingMenu);
+// } else {
+//     SystrayMenu.Items.push(MessagingMenu);
+// }
 
-export default MessagingMenuWidget;
+const systrayItem = {
+    Component: MessagingMenu,
+};
+registry.category('systray').add(
+    'mail.messaging_menu',
+    systrayItem,
+    { sequence: 10 },
+);
```

Comment:
The register of messaging menu component should not be on the old widget file, but on the boot file/messaging service file.